packages/components/src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,215 @@
+  @Prop() public indeterminate?: boolean = false;
+
+  /** Marks the checkbox as pre-selected (checked) on initial load. */
+  @Prop() public checked?: boolean = false;
+
+  /** The id of a form element the checkbox should be associated with. */
+  @Prop() public form?: string;
+
+  /**
+   * The checkbox value.
+   * When a form is submitted, only a checkbox which is currently checked is included in the submission.
+   */
+  @Prop() public value?: string = 'on';
+
+  /** The label text. */
+  @Prop() public label?: string = '';
+
+  /** The validation state. */
+  @Prop() public state?: CheckboxState = 'none';
+
+  /** The message styled depending on validation state. */
+  @Prop() public message?: string = '';
+
+  /** Show or hide label. For better accessibility it's recommended to show the label. */
+  @Prop() public hideLabel?: BreakpointCustomizable<boolean> = false;
+
+  /** @experimental Disables the checkbox and shows a loading indicator. */
+  @Prop() public loading?: boolean = false;
+
+  /** Adapts the color depending on the theme. */
+  @Prop() public theme?: Theme = 'light';
+
+  /** Emitted when checkbox checked property is changed. */
+  @Event({ bubbles: false }) public update: EventEmitter<CheckboxUpdateEventDetail>;
+
+  @AttachInternals() private internals: ElementInternals;
+
+  private controllerHost = new ControllerHost(this);
+  private loadingCtrl = new InitialLoadingController(this.controllerHost);
+  private checkboxInputElement: HTMLInputElement;
+
+  @Listen('keydown')
+  public onKeydown(e: KeyboardEvent): void {
+    const { key } = e;
+    if ((key === ' ' || key === 'Spacebar') && isDisabledOrLoading(this.disabled, this.loading)) {
+      e.preventDefault();
+    }
+  }
+
+  @Watch('value')
+  public onValueChange(newValue: string): void {
+    this.internals.setFormValue(this.checkboxInputElement && this.checkboxInputElement.checked ? newValue : undefined);
+  }
+
+  @Watch('indeterminate')
+  public onIndeterminateChange(newValue: boolean): void {
+    if (this.checkboxInputElement) {
+      this.checkboxInputElement.indeterminate = newValue;
+    }
+  }
+
+  @Watch('checked')
+  public onCheckedChange(newValue: boolean): void {
+    this.internals.setFormValue(newValue ? this.value : undefined);
+  }
+
+  public connectedCallback(): void {
+    applyConstructableStylesheetStyles(this.host, getSlottedAnchorStyles, getCheckboxRadioButtonSafariRenderingFix);
+  }
+
+  public componentShouldUpdate(newVal: unknown, oldVal: unknown): boolean {
+    return hasPropValueChanged(newVal, oldVal);
+  }
+
+  public componentDidLoad(): void {
+    this.checkboxInputElement.indeterminate = this.indeterminate;
+    if (this.checkboxInputElement.checked) {
+      this.internals.setFormValue(this.value);
+    }
+  }
+
+  public formResetCallback(): void {
+    this.internals.setValidity({});
+    this.internals.setFormValue(undefined);
+    this.checkboxInputElement.checked = false;
+    this.checked = false;
```

Comment:
```suggestion
    this.checked = false;
```

Shoudn't the `checkboxInputElement` checked value automatically get updated when the component rerenders?